go.mod

```diff
@@ -6,7 +6,7 @@ replace launchpad.net/gocheck => github.com/go-check/check v0.0.0-20180628173108
 
 require (
 	cloud.google.com/go v0.63.0 // indirect
-	github.com/container-storage-interface/spec v1.1.0
+	github.com/container-storage-interface/spec v1.3.0 // indirect
```

Comment:
This was added when I run "make test".
(not e2e)

Reply:
That's because we are using a very old `csi-test` package.
We should update it to v4 at the same time.

Here is my suggestion:
```diff
diff --git a/e2e/suite_test.go b/e2e/suite_test.go
index e6c8295..bb6c3f2 100644
--- a/e2e/suite_test.go
+++ b/e2e/suite_test.go
@@ -9,7 +9,7 @@ import (
        "testing"
        "time"

-       "github.com/kubernetes-csi/csi-test/pkg/sanity"
+       "github.com/kubernetes-csi/csi-test/v4/pkg/sanity"
        . "github.com/onsi/ginkgo"
        . "github.com/onsi/gomega"
        appsv1 "k8s.io/api/apps/v1"
@@ -134,13 +134,13 @@ var _ = Describe("TopoLVM", func() {
                        }).Should(Succeed())
                })

-               sanity.GinkgoTest(&sanity.Config{
-                       Address:           "/tmp/topolvm/worker1/plugins/topolvm.cybozu.com/node/csi-topolvm.sock",
-                       ControllerAddress: "/tmp/topolvm/worker1/plugins/topolvm.cybozu.com/controller/csi-topolvm.sock",
-                       TargetPath:        "/tmp/topolvm/worker1/plugins/topolvm.cybozu.com/node/mountdir",
-                       StagingPath:       "/tmp/topolvm/worker1/plugins/topolvm.cybozu.com/node/stagingdir",
-                       TestVolumeSize:    1073741824,
-                       IDGen:             &sanity.DefaultIDGenerator{},
-               })
+        tc := sanity.NewTestConfig()
+        tc.Address = "/tmp/topolvm/worker1/plugins/topolvm.cybozu.com/node/csi-topolvm.sock"
+        tc.ControllerAddress = "/tmp/topolvm/worker1/plugins/topolvm.cybozu.com/controller/csi-topolvm.sock"
+        tc.TargetPath = "/tmp/topolvm/worker1/plugins/topolvm.cybozu.com/node/mountdir"
+        tc.StagingPath = "/tmp/topolvm/worker1/plugins/topolvm.cybozu.com/node/stagingdir"
+        tc.TestVolumeSize = 1073741824
+        tc.IDGen = &sanity.DefaultIDGenerator{}
+               sanity.GinkgoTest(&tc)
        })
 })
diff --git a/go.mod b/go.mod
index 35c344e..26bfeee 100644
--- a/go.mod
+++ b/go.mod
@@ -6,13 +6,12 @@ replace launchpad.net/gocheck => github.com/go-check/check v0.0.0-20180628173108

 require (
        cloud.google.com/go v0.63.0 // indirect
-       github.com/container-storage-interface/spec v1.3.0 // indirect
        github.com/cybozu-go/log v1.5.0
        github.com/cybozu-go/well v1.10.0
        github.com/go-logr/logr v0.1.0
        github.com/golang/protobuf v1.4.2
        github.com/google/go-cmp v0.5.2
-       github.com/kubernetes-csi/csi-test v2.2.0+incompatible
+       github.com/kubernetes-csi/csi-test/v4 v4.0.2
        github.com/onsi/ginkgo v1.14.1
        github.com/onsi/gomega v1.10.2
        github.com/prometheus/client_golang v1.1.0
```

Reply:
Thank you. I updated csi-test package version.